src/routes/solid-start/guides/actions-and-mutations.mdx

```diff
@@ -0,0 +1,606 @@
+---
+title: "Actions and Mutations"
+---
+
+This guide provides practical examples of using actions to mutate data in SolidStart.
+
+## Handling form submission
+
+To handle [`<form>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form) submissions with an action:
+
+1. Ensure the action has a unique name. [Learn more here](/solid-router/reference/data-apis/action#notes-of-form-implementation-and-ssr).
+2. Pass the action to the `<form>` element using the `action` prop.
+3. Ensure the `<form>` element uses the `post` method for submission.
+4. Use the [`FormData`](https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData) object in the action to extract field data using the navite `FormData` methods.
+
+<TabsCodeBlocks>
+<div id="TypeScript">
+
+```tsx {3-9} {13} title="src/routes/index.tsx"
+import { action } from "@solidjs/router";
+
+const addPost = action(async (formData: FormData) => {
+	const title = formData.get("title");
```

Comment:
this will raise ~~alarms~~ squiggles in the TypeScript LSP because a form value can be `string`, `number`, `File`, or `null`.

The user will either need to add a type guard or a type assertion here.
Though I generally consider type assertion a code smell, it's quite a harmless one in this specific scenario and we need to consider the focus of this code snippet.

```suggestion
	const title = formData.get("title") as string;
```